lib/hyperion/types/hyperion_uri.rb

```diff
@@ -0,0 +1,97 @@
+require 'uri'
+require 'delegate'
+require 'active_support/core_ext/hash/keys'
+require 'rack/utils'
+require 'abstractivator/array_ext'
+
+class HyperionUri < SimpleDelegator
+  # An enhanced version of URI. Namely,
+  # - the base uri is a first class citizen,
+  # - it accepts a hash containing query key/values, and
+  # - it form-encodes query values that are arrays.
+
+  attr_accessor :query_hash
+
+  def initialize(uri, query_hash={})
+    @uri = make_ruby_uri(uri)
+    query_from_uri = parse_query(@uri.query)
+    additional_query_params = validate_query(query_hash)
+    @query_hash = query_from_uri.merge(additional_query_params)
+    __setobj__(@uri)
+  end
+
+  def query
+    query_string(@query_hash)
+  end
+
+  def query=(query)
+    @query_hash = parse_query(query)
+  end
+
+  def to_s
+    fixed = @uri.dup
+    fixed.query = query
+    make_ruby_uri(fixed).to_s
+  end
+
+  def inspect
+    "#<HyperionUri:0x#{(object_id << 1).to_s(16)} #{to_s}>"
+  end
+
+  # @return [String] the URI base e.g., "h\ttp://somehost:80"
```

Comment:
iirc, if the backslash is not in there, it shows up as link to http://somehost:80 with text "somehost"